pkg/envoy/json_test.go

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of Cilium
+
+//go:build !privileged_tests
+// +build !privileged_tests
+
+package envoy
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+
+	cilium_v2alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	envoy_config_http "github.com/cilium/proxy/go/envoy/extensions/filters/network/http_connection_manager/v3"
+
+	. "gopkg.in/check.v1"
+)
+
+type JSONSuite struct{}
```

Comment:
We have an open issue to move to the standard Go testing framework, rather than to add new tests using the old go-check framework: https://github.com/cilium/cilium/issues/16860

Reply:
Would be nice to have a conversion guide, i.e., how to do comparable asserts for `IsNil`, `HasLen`, `Equals` (for simpler and deeper types). Until we have that I'm inclined not to invest the time right now on converting the tests.

Reply:
Fair enough. FWIW, the tests would be converted to use testify/assert which has a pretty straightforward set of helpers to do assertions: https://pkg.go.dev/github.com/stretchr/testify/assert